src/site/content/en/reliable/resilient-search-experiences/index.md

```diff
@@ -0,0 +1,100 @@
+
+```javascript
+import {BackgroundSyncPlugin} from 'workbox-background-sync'; 
+import {registerRoute} from 'workbox-routing'; 
+import {NetworkOnly} from 'workbox-strategies';
+```
+
+Next, create an instance of the [workbox.backgroundSync plugin](https://developers.google.com/web/tools/workbox/reference-docs/v4/workbox.backgroundSync.Plugin), to automatically add failed requests to a queue, so they can be retried later:
+
+```javascript
+const bgSyncPlugin = new workbox.backgroundSync.Plugin('offlineQueryQueue', {
+  maxRetentionTime: 60,
+  onSync: async ({queue}) => {
+    let entry;
+    while ((entry = await queue.shiftRequest())) {
+      try {
+        const response = await fetch(entry.request);
+        const cache = await caches.open('offline-search-responses');
+        const offlineUrl = entry.request.url + '&notification=true';
+        cache.put(offlineUrl, response);
+        showNotification(offlineUrl);
+      } catch (error) {
+        await this.unshiftRequest(entry);
+        throw error;
+      }
+    }
+  },
+});
+```
+
+The plugin receives the following parameters:
+
+- `offlineQueryQueue`: The name of the queue that will be used to persist the failed requests in [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API).
+- `maxRetentionTime`: The amount of time in minutes a request may be retried, after which point they will be discarded.
+- `onSync`: The callback that will be triggered when the connection is recovered. At that point, each failed request can be dequeued and processed, by calling `queue.shiftRequest()`.
+
+Finally, define a [networkOnly](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#network_only) runtime caching strategy for requests to the search URL (e.g. `/search_action`) and pass it the `bgSyncPlugin` defined previously:
+
+```javascript
+workbox.routing.registerRoute(
+  matchSearchUrl,
+  new workbox.strategies.NetworkOnly({
+        plugins: [bgSyncPlugin]
+  })
+);
+```
+
+This tells Workbox to always go to the network when the service worker intercepts a request for the search endpoint, and to delegate to the Background Sync plugin the task of managing offline scenarios.
+
+As a result, when the user goes offline while searching, the query is automatically saved. When the connection is recovered the offline logic is triggered to process the request and inform the user of the result.
+
+## Conclusion
+
+In this article you learned how to implement a search experience capable of responding gracefully to offline scenarios, by combining the [Background Sync API](https://developers.google.com/web/updates/2015/12/background-sync) and the [Push API](https://developer.mozilla.org/en-US/docs/Web/API/Push_API).
+We used Workbox to show how to implement this feature, because it's our recommended way of building web apps that use service workers, but the same can be achieved by writing vanilla service worker code.
```

Comment:
```suggestion
We used Workbox to show how to implement this feature, because it's Google's recommended way of building web apps that use service workers, but the same can be achieved by writing vanilla service worker code.
```
@jeffposnick can we say this or is it too much?

Reply:
I don't know if we need to make the claim that "Google" recommends usage of Workbox. (Folks should use whatever makes the most sense for them.)

How about:

> We used Workbox to show how to implement this feature, as it simplifies the process, but the same can be achieved by writing vanilla service worker code.

Reply:
This suggestin LGTM! Let's see what @kaycebasques has to say 👍

Reply:
Jeff's rewrite LGTM